pkg/plugins/builtin/callback/models.go

```diff
@@ -0,0 +1,169 @@
+package plugincallback
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/Masterminds/squirrel"
+	"github.com/gofrs/uuid"
+	"github.com/juju/errors"
+	"github.com/loopfz/gadgeto/zesty"
+	"github.com/ovh/utask/db/pgjuju"
+	"github.com/ovh/utask/db/sqlgenerator"
+	"github.com/ovh/utask/models"
+	"github.com/ovh/utask/models/resolution"
+	"github.com/ovh/utask/models/task"
+	"github.com/ovh/utask/pkg/jsonschema"
+	"github.com/ovh/utask/pkg/now"
+)
+
+type callback struct {
+	ID               int64           `json:"-" db:"id"`
+	PublicID         string          `json:"id" db:"public_id"`
+	TaskID           int64           `json:"-" db:"id_task"`
+	ResolutionID     int64           `json:"-" db:"id_resolution"`
+	Created          time.Time       `json:"created" db:"created"`
+	Updated          time.Time       `json:"updated" db:"updated"`
+	Called           *time.Time      `json:"called" db:"called"`
+	ResolverUsername string          `json:"resolver" db:"resolver_username"`
+	EncryptedSchema  []byte          `json:"-" db:"encrypted_schema"`
+	EncryptedSecret  []byte          `json:"-" db:"encrypted_secret"`
+	EncryptedBody    []byte          `json:"-" db:"encrypted_body"`
+	Secret           string          `json:"-" db:"-"`
+	Body             json.RawMessage `json:"body" db:"-"`
+	Schema           json.RawMessage `json:"schema" db:"-"`
+}
+
+func createCallback(dbp zesty.DBProvider, task *task.Task, ctx *CallbackContext, schemaJSON string) (cb *callback, err error) {
+	defer errors.DeferredAnnotatef(&err, "Failed to create callback")
+
+	resolution, err := resolution.LoadFromPublicID(dbp, *task.Resolution)
+	if err != nil {
+		return nil, err
+	}
+
+	cb = &callback{
+		PublicID:         uuid.Must(uuid.NewV4()).String(),
+		TaskID:           task.ID,
+		ResolutionID:     resolution.ID,
+		Created:          now.Get(),
+		Updated:          now.Get(),
+		ResolverUsername: ctx.RequesterUsername,
+		Secret:           uuid.Must(uuid.NewV4()).String(),
+	}
+
+	if schemaJSON != "" {
+		cb.Schema, err = jsonschema.NormalizeAndCompile(ctx.StepName, json.RawMessage(schemaJSON))
+
+		if err != nil {
+			return nil, errors.NewBadRequest(fmt.Errorf("unable to parse provided schema: %s", err), "")
+		}
+	}
+
+	cb.EncryptedSchema, err = models.EncryptionKey.Encrypt(cb.Schema, []byte(cb.PublicID))
```

Comment:
As we have encrypted data here, we should also modify https://github.com/ovh/utask/blob/master/api/server.go#L529-L538 to include a rollover of the callbacks data when the encryption key is changing.